src/app/todo/services/task-api.service.ts

```diff
@@ -0,0 +1,36 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { ITask } from '../models/todo.interface';
+import { Observable, catchError, throwError, map } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TaskApiService {
+  constructor(private http: HttpClient) {}
+  get(url: string): Observable<ITask[]> {
+    return this.http.get<ITask[]>(url);
+  }
+  getTask(url: string, id: number): Observable<ITask> {
```

Comment:
As mentioned before you can move these to 1 service TaskService and also what is get and post methods used for? Confusing naming for those 2